### PR TITLE
[Compute] vmss create/update: add --enable-terminate-notification (only for update) and --terminate-notification-time parameters to support terminate scheduled event configurability. Issue #10124

### PR DESCRIPTION
**Pull request by qwordy**
[Compute] vmss create/update: add --enable-terminate-notification (only for update) and --terminate-notification-time parameters to support terminate scheduled event configurability. Issue #10124
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by qwordy** [comment]
Original PR: #10391 
I have to open a new PR due to some mistakes.

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py [hunk 1]
```diff
@@ -795,6 +796,15 @@ def build_vmss_resource(cmd, name, naming_prefix, location, tags, overprovision,
     if proximity_placement_group:
         vmss_properties['proximityPlacementGroup'] = {'id': proximity_placement_group}
 
+    if terminate_notification_time is not None:
+        scheduled_events_profile = {
+            'terminateNotificationProfile': {
+                'notBeforeTimeout': terminate_notification_time,
```
Comment (szimmerMicrosoft) [inline src/azure-cli/azure/cli/command_modules/vm/_template_builder.py]: Something to consider.

In the powershell command line implementation they just allow users to specify the time in minutes, and the powerhsell cmdlet converts to the appropraite time.

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -21,6 +21,12 @@ Release History
 * [BREAKING] When not specified, the default value for `--start-task-wait-for-success` on `az batch pool create` is now true (was false).
 * [BREAKING] The default value for Scope on AutoUserSpecification is now always Pool (was Task on Windows nodes, Pool on Linux nodes). This argument is not exposed via the commandline, but can be set in the `--json-file` arguments.
 
+**Compute**
```
Comment (szimmerMicrosoft) [inline src/azure-cli/HISTORY.rst]: What is the output of the vmss show command?
Is it the json output of the response?

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_template_builder.py]: > 
> 
> Something to consider.
> 
> In the powershell command line implementation they just allow users to specify the time in minutes, and the powerhsell cmdlet converts to the appropraite time.

Good idea! ISO 8601 format is complicated!

File: src/azure-cli/HISTORY.rst (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/HISTORY.rst]: > 
> 
> What is the output of the vmss show command?
> Is it the json output of the response?


`vmss show` visit `GET https://management.azure.com/subscriptions/{...}/resourceGroups/{...}/providers/Microsoft.Compute/virtualMachineScaleSets/{name}?api-version=2019-03-01`.

Basically, Yes.

Comment (qwordy) [reply src/azure-cli/HISTORY.rst]: And, `vmss get-instance-view` visit `GET https://management.azure.com/subscriptions/{...}/resourceGroups/{...}/providers/Microsoft.Compute/virtualMachineScaleSets/{name}/instanceView?api-version=2019-03-01`

Comment (qwordy) [reply src/azure-cli/HISTORY.rst]: What's the difference between `virtualMachineScaleSets/{name}?$expand=instanceView` and `virtualMachineScaleSets/{name}/instanceView`?

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 3]
```diff
@@ -500,6 +500,13 @@ def load_arguments(self, _):
 
         c.argument('protect_from_scale_in', arg_type=protection_policy_type, help="Protect the VM instance from scale-in operations.")
         c.argument('protect_from_scale_set_actions', arg_type=protection_policy_type, help="Protect the VM instance from scale set actions (including scale-in).")
+        c.argument('enable_terminate_notification', min_api='2019-03-01', arg_type=get_three_state_flag(),
+                   help='Enable terminate notification')
+
+    for scope in ['vmss create', 'vmss update']:
+        with self.argument_context(scope) as c:
+            c.argument('terminate_notification_time', min_api='2019-03-01',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: Can you provide default value for your argument?

**Comment by MyronFanQiu** [comment]
just one question. Does it support the time more than 60?

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 3 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: > 
> 
> Can you provide default value for your argument?

The expected default behaviour is disabled. When users don't set this argument, it is disabled.

**Comment by qwordy** [comment]
> 
> 
> just one question. Does it support the time more than 60 minutes?



> 
> 
> just one question. Does it support the time more than 60 minutes?

Server has limitation. 5~15 min.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 4]
```diff
@@ -500,6 +500,13 @@ def load_arguments(self, _):
...
+                       help='Length of time (in minutes) a notification to be sent to the VM on the instance metadata server till the VM gets deleted')
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: I prefer adding the limitation information into help without checking it.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: > 
> 
> I prefer adding the limitation information into help without checking it.

Good idea!

File: src/azure-cli/HISTORY.rst (hunk 2 above)
Comment (szimmerMicrosoft) [reply src/azure-cli/HISTORY.rst]: virtualMachineScaleSets/{name}?$expand=instanceView returns the vmScaleSet goal state with a property named instance view populated (instanceView is expensive runtime state)

virtualMachineScaleSets/{name}/instanceView just returns the expensive runtimestate without the vmScaleSet goal state infromation

**Pull request merged**